lib/Dialect/SSP/CMakeLists.txt

```diff
@@ -11,6 +11,7 @@ set(SSP_Srcs
   )
 
 set(SSP_LinkLibs
+  CIRCTScheduling
```

Comment:
I'm unsure about this library dep. Technically, the only ties that the dialect implementation has into `circt::scheduling` are the additional members on the attribute classes, calling into the `Problem` classes' getters and setters, which are defined in the header (`Problems.h`).

Any component using the dialect to import/export problems (e.g. the upcoming test passes) is going to link `CIRCTScheduling` anyways because it's interacting with `Problem`s.